components/kyma-environment-broker/common/orchestration/resolver.go

```diff
@@ -141,21 +141,15 @@ func (resolver *GardenerRuntimeResolver) resolveRuntimeTarget(rt RuntimeTarget,
 			resolver.logger.Errorf("Failed to get runtimeID from %s annotation for Shoot %s", runtimeIDAnnotation, shoot.Name)
 			continue
 		}
-		runtime, ok := resolver.getRuntime(runtimeID)
+		r, ok := resolver.getRuntime(runtimeID)
 		if !ok {
 			resolver.logger.Errorf("Couldn't find runtime for runtimeID %s", runtimeID)
 			continue
 		}
-		provState := ""
-		deprovState := ""
-		if runtime.Status.Provisioning != nil {
-			provState = runtime.Status.Provisioning.State
-		}
-		if runtime.Status.Deprovisioning != nil {
-			deprovState = runtime.Status.Deprovisioning.State
-		}
-		if provState != string(brokerapi.Succeeded) || deprovState != "" {
-			resolver.logger.Infof("Skipping Shoot %s (runtimeID: %s, instanceID %s) due to provisioning/deprovisioning state: %s/%s", shoot.Name, runtimeID, runtime.InstanceID, provState, deprovState)
+
+		lastOp, lastOpType := runtime.FindLastOperation(r)
+		if lastOpType == runtime.Deprovision || lastOpType == runtime.Suspension || lastOpType == runtime.Provision && lastOp.State != string(brokerapi.Succeeded) {
```

Comment:
shouldn't we also skip shoots with the unsuspension operation type which weren't succeeded?